compiler/tests/testSuite.bal

```diff
@@ -37,6 +44,8 @@ function testCompileEU(string path, string kind) returns file:Error|io:Error? {
             test:assertNotExactEquals(err, (), "expected an error " + path);
         }
         else {
+            // JBUG #31334 cast needed
+            verifyErrorFileName(err);
```

Comment:
You need to look at the .modules directory here and find the file and line number of the @error line

Reply:
Changed such that it'll check if there is a file in .modules directory matching the error's filename and optimistically accept that is to be the file that caused the error. This is because if A imports B we will get the error in A as well with B's name (but not path)

Reply:
> This is because if A imports B we will get the error in A as well with B's name (but not path)

Can you explain more? That seems strange. Which test case gives you exactly what error?

Reply:
There's a bug with how we are using s:defnLocation. It gives us the line number of definition, but we are using it where we only want the filename. Maybe this is related.

Reply:
What I meant was If lets say `foo.bal` import `bar.bal` and `bar.bal` has an error, error's filename is just `bar.bal` not the path to `bar.bal` (unlike if `foo.bal` had the error in which case you get a path not just the filename). So then what I am doing is going over all the files in foo.bal's modules to see if I can find a `bar.bal` (this is because I don't know to what sub module `bar.bal` belongs to from the error). A problem will arise if we have multiple `bar.bal` files because we can't determine which `bar.bal` with the current error information.

Reply:
> error's filename is just bar.bal 

That doesn't sound right.  Which test case in your pull request is this?

Reply:
fixed by making imported files return path relative to modules directory